gazebo_plugins/src/gazebo_ros_imu_sensor.cpp

```diff
@@ -0,0 +1,255 @@
+    imu_msg.linear_acceleration.y = accelerometer_data.y + GuassianKernel(0,gaussian_noise);
+    imu_msg.linear_acceleration.z = accelerometer_data.z + GuassianKernel(0,gaussian_noise);
+
+    imu_msg.angular_velocity.x = gyroscope_data.x + GuassianKernel(0,gaussian_noise);
+    imu_msg.angular_velocity.y = gyroscope_data.y + GuassianKernel(0,gaussian_noise);
+    imu_msg.angular_velocity.z = gyroscope_data.z + GuassianKernel(0,gaussian_noise);
+
+    //covariance is related to the Gaussian noise
+    double gn2 = gaussian_noise*gaussian_noise;
+    imu_msg.orientation_covariance[0] = gn2;
+    imu_msg.orientation_covariance[4] = gn2;
+    imu_msg.orientation_covariance[8] = gn2;
+    imu_msg.angular_velocity_covariance[0] = gn2;
+    imu_msg.angular_velocity_covariance[4] = gn2;
+    imu_msg.angular_velocity_covariance[8] = gn2;
+    imu_msg.linear_acceleration_covariance[0] = gn2;
+    imu_msg.linear_acceleration_covariance[4] = gn2;
+    imu_msg.linear_acceleration_covariance[8] = gn2;
+
+    //preparing message header
+    imu_msg.header.frame_id = body_name;
+    imu_msg.header.stamp.sec = current_time.sec;
+    imu_msg.header.stamp.nsec = current_time.nsec;
+
+    //publishing data
+    imu_data_publisher.publish(imu_msg);
+
+    ros::spinOnce();
+  }
+
+  last_time = current_time;
+}
+
+double gazebo::GazeboRosImuSensor::GuassianKernel(double mu, double sigma)
+{
+  // generation of two normalized uniform random variables
+  double U1 = static_cast<double>(rand_r(&seed)) / static_cast<double>(RAND_MAX);
+  double U2 = static_cast<double>(rand_r(&seed)) / static_cast<double>(RAND_MAX);
+
+  // using Box-Muller transform to obtain a varaible with a standard normal distribution
+  double Z0 = sqrt(-2.0 * ::log(U1)) * cos(2.0*M_PI * U2);
+
+  // scaling
+  Z0 = sigma * Z0 + mu;
+  return Z0;
+}
+
+bool gazebo::GazeboRosImuSensor::LoadParameters()
+{
+  //loading parameters from the sdf file
+
+  //NAMESPACE
+  if (sdf->HasElement("robotNamespace"))
+  {
+    robot_namespace =  "/" + sdf->Get<std::string>("robotNamespace") +"/";
+    ROS_INFO_STREAM("<robotNamespace> set to: "<<robot_namespace);
+  }
+  else
+  {
+#if GAZEBO_MAJOR_VERSION >= 7
+    std::string scoped_name = sensor->ParentName();
+#else
+    std::string scoped_name = sensor->GetParentName(); //HACK to find the model name
+#endif
+    std::size_t it = scoped_name.find("::");
+
+    robot_namespace = "/" +scoped_name.substr(0,it)+"/";
+    ROS_WARN_STREAM("missing <robotNamespace>, set to default: " << robot_namespace);
+  }
+
+  //TOPIC
+  if (sdf->HasElement("topicName"))
+  {
+    topic_name =  robot_namespace + sdf->Get<std::string>("topicName");
+    ROS_INFO_STREAM("<topicName> set to: "<<topic_name);
+  }
+  else
+  {
+    topic_name = robot_namespace + "/imu_data";
+    ROS_WARN_STREAM("missing <topicName>, set to /namespace/default: " << topic_name);
+  }
+
+  //BODY NAME
+  if (sdf->HasElement("bodyName"))
```

Comment:
"bodyName" is referenced as "frameName" in other plugins (see [here](http://gazebosim.org/tutorials?tut=ros_gzplugins&cat=connect_ros#Camera)), maybe better keeping the same convention ?